### PR TITLE
core, server: refine logging

### LINKED ISSUE #1050: replace fern with tracing on the server

### PR DESCRIPTION
**Pull request by Parth**
+ fixes #1050

### CODE REVIEW

File: server/server/src/billing/billing_model.rs [hunk 1]
```diff
@@ -45,10 +45,7 @@ impl BillingPlatform {
         let expiration_time = expiry_information
             .expiry_time_millis
             .ok_or_else::<ServerError<UpgradeAccountGooglePlayError>, _>(|| {
-                internal!(
-                    "Cannot get expiration time of a recovered subscription. public_key {:?}",
-                    &public_key
```
Comment (steverusso) [inline server/server/src/billing/billing_model.rs:50]: Not using `public_key` makes it an unused variable

File: server/server/src/router_service.rs [hunk 2]
```diff
@@ -8,12 +8,13 @@ use lazy_static::lazy_static;
...
 use lockbook_models::api::*;
 use lockbook_models::api::{ErrorWrapper, Request, RequestWrapper};
-use log::{error, warn};
 use prometheus::{register_histogram_vec, HistogramVec, TextEncoder};
 use serde::de::DeserializeOwned;
 use serde::Serialize;
 use std::collections::HashMap;
 use std::sync::Arc;
+use tracing::*;
+use tracing::{span, Level};
```
Comment (steverusso) [inline server/server/src/router_service.rs:17]: I would just chose one style, either `*` or specify, but at a minimum the second line is unnecessary.

File: server/server/src/router_service.rs [hunk 3]
```diff
@@ -34,7 +35,9 @@ macro_rules! core_req {
         use crate::router_service::{deserialize_and_check, method};
         use crate::{RequestContext, ServerError};
         use lockbook_models::api::{ErrorWrapper, Request};
-        use log::error;
+        use lockbook_models::file_metadata::Owner;
+        use tracing::*;
+        use tracing::{span, Level};
```
Comment (steverusso) [inline server/server/src/router_service.rs:40]: Same as above.

**Review by steverusso — CHANGES_REQUESTED** [review]

File: server/server/src/router_service.rs (hunk 3 above)
Comment (Parth) [reply server/server/src/router_service.rs]: https://github.com/lockbook/lockbook/pull/1211/commits/1525f90016a8282dc8ce97d39933451113983e32

**Review by steverusso — APPROVED** [review]

**Pull request merged**